### ISSUE TITLE
GraphQL Field Collection for avoid dataloaders and solve N+1 problem differently

### ISSUE THREAD

**Issue by frederikhors** [async-graphql/async-graphql#963]
Hi all! :smiley:

I'm using [gqlgen](https://github.com/99designs/gqlgen) (like async-graphql but in Go not in Rust) and [Ent](https://entgo.io/) as ORM (but is more than an ORM).

I'm writing to you because I just found out an excellent feaure of Ent: [GraphQL Field Collection](https://entgo.io/docs/tutorial-todo-gql-field-collection).

Since I found out SeaORM and their amazing [summer project](https://github.com/SeaQL/summer-of-code/discussions/12) [I asked them](https://github.com/SeaQL/summer-of-code/discussions/12#discussioncomment-3021867) if would be possibile to add this feature on SeaORM too when using it with async-graphql.

Basically Ent with this feature offers the possibility to AVOID ALL THE DATALOADERs.

Do you think async-graphql offers the ability to do that?

**Can we use some async-graphql "metadata" to instruct SeaORM (or others) to detect what edges are being requested in queries?**

_I still don't know async-graphql, excuse the maybe silly question. I am preparing for my - hopefully very close - switch to Rust._

Thank you for your work.

**Comment by Miaxos** [comment]
Hello!

>I'm writing to you because I just found out an excellent feaure of Ent: [GraphQL Field Collection](https://entgo.io/docs/tutorial-todo-gql-field-collection).

There example is flawed, with async-graphql & dataloader you can do the same as what they are doing with only 2 queries too, the first one get the list then a dataloader collect and execute a query to get the parent of each one.

>Do you think async-graphql offers the ability to do that?

Short answer: possible but really tricky.

Well, it should be possible, it's a query engine, but it's really tricky because you would need to process the query a first time to pre-construct the queries you would need to execute and when you "resolve" the actual fields it would need to get data from the pre-constructed query.

The first issue I'm seeing it to be able to operate that kind of gql schema, and to create a query engine coupled with your db schema, you'll need to generate the associated graphql code from the db schema.

**Comment by frederikhors** [comment]
Nope. I'm still studying... Please. I'll close it.

**Comment by frederikhors** [comment]
Nope

**Comment by sunli829** [comment]
I found a new project, maybe you can refer to it. 🙂

https://github.com/SeaQL/seaography
